packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -89,86 +58,25 @@ class InputDateTimePicker extends React.Component {
 		super(props);
 
-		if (!warnOnce.unstable) {
-			// eslint-disable-next-line
-			console.warn(
-				"UNSTABLE WARNING: The 'InputDateTimePicker' and all the sub components aren't ready to be used in Apps. Code can (will) change outside the release process until it's ready.",
-			);
-			warnOnce.unstable = true;
-		}
-
-		checkSupportedDateFormat(props.dateFormat);
 		this.popoverId = `date-time-picker-${props.id || uuid.v4()}`;
-		this.inputErrorId = `${props.id}-input-error`;
-		this.hoursErrorId = `${props.id}-hours-error`;
-		this.minutesErrorId = `${props.id}-minutes-error`;
-		this.secondsErrorId = `${props.id}-seconds-error`;
-		this.initialState = extractParts(props.selectedDateTime, this.getDateOptions());
 		this.state = {
-			...this.initialState,
 			showPicker: false,
-			previousErrors: [],
 		};
 
-		this.onInputFocus = this.onInputFocus.bind(this);
-		this.hasError = this.hasError.bind(this);
 		this.onBlur = this.onBlur.bind(this);
 		this.onFocus = this.onFocus.bind(this);
-		this.onSubmit = this.onSubmit.bind(this);
-		this.onInputChange = this.onInputChange.bind(this);
-		this.onPickerChange = this.onPickerChange.bind(this);
+		this.onChange = this.onChange.bind(this);
 		this.onKeyDown = this.onKeyDown.bind(this);
 		this.openPicker = this.setPickerVisibility.bind(this, true);
 		this.closePicker = this.setPickerVisibility.bind(this, false);
 	}
 
-	componentWillReceiveProps(nextProps) {
-		const newSelectedDateTime = nextProps.selectedDateTime;
-
-		const needDateTimeStateUpdate =
-			newSelectedDateTime !== this.props.selectedDateTime && // selectedDateTime props updated
-			newSelectedDateTime !== this.state.datetime && // not the same ref as state date time
-			!isSameSecond(newSelectedDateTime, this.state.datetime); // not the same value as state
-
-		if (nextProps.dateFormat !== this.props.dateFormat) {
-			checkSupportedDateFormat(nextProps.dateFormat);
-		}
-
-		if (needDateTimeStateUpdate) {
-			const dateRelatedPartState = extractParts(newSelectedDateTime, this.getDateOptions());
-			this.setState(dateRelatedPartState);
-		}
-	}
-
-	onChange(event, origin) {
-		const { errorMessage, datetime, textInput, errors, previousErrors } = this.state;
-
-		if (
-			this.props.onChange &&
-			(this.dateHasChanged() || haveErrorsChanged(previousErrors, errors))
-		) {
-			// we need to update the initial state once it has been changed
-			this.initialState = { ...this.state };
-			this.props.onChange(event, { errors, errorMessage, datetime, textInput, origin });
-		}
-	}
-
-	onInputChange(event) {
-		const textInput = event.target.value;
-		const nextState = extractPartsFromTextInput(textInput, this.getDateOptions());
-		this.setState({ previousErrors: this.state.errors, ...nextState }, () => {
-			if (!this.props.formMode) {
-				this.onChange(event, 'INPUT');
-			}
-		});
-	}
-
-	onKeyDown(event) {
+	onKeyDown(event, { onReset }) {
 		switch (event.keyCode) {
 			case keycode.codes.esc:
+				onReset();
```

Comment:
is it possible to have blur called here ? 
In the grid case we need this event to be called, click outside and escape should work the same